nvflare/lighter/entity.py

```diff
@@ -355,114 +407,165 @@ def __init__(
+            participants = []
+            self._participants_by_types[participant.type] = participants
+        participants.append(participant)
+        self._all_names[participant.name] = participant
+        return participant
 
-        result.extend(self.clients)
-        result.extend(self.relays)
-        result.extend(self.admins)
+    def add_client(self, name: str, org: str, props: dict) -> Participant:
+        """Add a client to the project
+
+        Args:
+            name: name of the client
+            org: org of the client
+            props: additional properties of the client
+
+        Returns: the Participant object of the client
+
+        """
+        return self.add_participant(Participant(ParticipantType.CLIENT, name, org, props))
+
+    def get_clients(self) -> List[Participant]:
+        """Get all clients of the project
+
+        Returns: a list of clients
+
+        """
+        return self.get_all_participants(ParticipantType.CLIENT)
+
+    def add_relay(self, name: str, org: str, props: dict) -> Participant:
+        """Add a relay to the project
+
+        Args:
+            name: name of the relay
+            org: org of the relay
+            props: additional properties of the relay
+
+        Returns: the relay Participant object
+
+        """
+        return self.add_participant(Participant(ParticipantType.RELAY, name, org, props))
+
+    def get_relays(self) -> List[Participant]:
+        """Get all relays of the project
+
+        Returns: the list of relays of the project
+
+        """
+        return self.get_all_participants(ParticipantType.RELAY)
+
+    def add_admin(self, name: str, org: str, props: dict) -> Participant:
+        """Add an admin user to the project
+
+        Args:
+            name: name of the admin user.
+            org: org of the admin user.
+            props: properties of the user definition
+
+        Returns: a Participant object of the admin user
+
+        """
+        return self.add_participant(Participant(ParticipantType.ADMIN, name, org, props))
+
+    def get_admins(self) -> List[Participant]:
+        """Get the list of admin users
+
+        Returns: list of admin users
+
+        """
+        return self.get_all_participants(ParticipantType.ADMIN)
+
+    def get_all_participants(self, types: Union[None, str, List[str]] = None):
+        """Get all participants of the project of specified types.
+
+        Args:
+            types: types of the participants to be returned.
+
+        Returns: all participants of the project of specified types.
+            If 'types' is not specified (None), it returns all participants of the project;
+            If 'types' is a str, it is treated as a single type and participants of this type is returned;
+            If 'types' is a list of types, participants of these types are returned;
+
+        """
+        if not types:
+            # get all types
+            return list(self._all_names.values())
```

Comment:
if the ``self._all_names = a={"clients": [1,2,3], "admin": [4,5]}``

this ``list(self._all_names.values())``

will give a nested list:

``[[1, 2, 3], [4, 5]]``

Do we want a flat list?
``[1,2,3,4,5]``